src/test/protocol/KnownFormatToGRPC_test.cpp

```diff
@@ -0,0 +1,924 @@
+    static std::map<std::string, SField const*>
+    soTemplateToSFields(
+        SOTemplate const& soTemplate,
+        [[maybe_unused]] KeyType fmtId)
+    {
+        std::map<std::string, SField const*> sFields;
+        for (SOElement const& element : soTemplate)
+        {
+            SField const& sField = element.sField();
+
+            // Fields that gRPC never includes.  This is probably intentional.
+            using FieldCode_t =
+                std::remove_const<decltype(SField::fieldCode)>::type;
+            static const std::set<FieldCode_t> excludedSFields{
+                sfLedgerIndex.fieldCode,
+                sfLedgerEntryType.fieldCode,
+                sfOperationLimit.fieldCode};
+
+            if (excludedSFields.count(sField.fieldCode))
+                continue;
+
+            // There are certain fields that gRPC never represents in
+            // transactions.  Exclude those.
+            if constexpr (std::is_same_v<KeyType, TxType>)
+            {
+                static const std::set<FieldCode_t> excludedTxFields{
+                    sfPreviousTxnID.fieldCode,
+                    sfTransactionType.fieldCode,
+                    sfWalletLocator.fieldCode,
+                    sfWalletSize.fieldCode};
+
+                if (excludedTxFields.count(sField.fieldCode))
+                    continue;
+            }
+
+            // If fmtId is a LedgerEntryType, exclude certain fields.
+            if constexpr (std::is_same_v<KeyType, LedgerEntryType>)
+            {
+                // Fields that gRPC does not include in certain LedgerFormats.
+                //
+                // Some of these may be bugs.
+                static const std::
+                    map<LedgerEntryType, std::vector<SField const*>>
+                        gRPCOmitFields{
+                            {ltACCOUNT_ROOT, {&sfWalletLocator, &sfWalletSize}},
+                            {ltDIR_NODE, {&sfExchangeRate}},
+                            {ltLEDGER_HASHES, {&sfFirstLedgerSequence}},
+                            {ltPAYCHAN, {&sfDestinationNode}},
+                        };
+
+                if (auto const iter = gRPCOmitFields.find(fmtId);
+                    iter != gRPCOmitFields.end())
+                {
+                    std::vector<SField const*> const& omits = iter->second;
+
+                    // Check for fields that gRPC omits from this type.
+                    if (std::find_if(
+                            omits.begin(),
+                            omits.end(),
+                            [&sField](SField const* const omit) {
+                                return *omit == sField;
+                            }) != omits.end())
+                    {
+                        // This is one of the fields that gRPC omits.
+                        continue;
+                    }
+                }
+            }
+
+            // The SFields and gRPC disagree on the names of some fields.
```

Comment:
I think we should explain why these differences exist, in the comment here. Most of these are the result of the gRPC API not abbreviating anything. 

`Fee -> XRPDropsAmount` is mainly the result of adding additional typing. 

There's the obvious misspelling with `TakerGetsCurreny`.

And then lastly, `Signature -> PaymentChannelSignature` was a deliberate deviation, with the rational being that `PaymentChannelClaim` transactions, when represented as JSON, have a `TxnSignature` field at the top level, and then a `Signature` field at a lower level. I thought this was confusing, so I changed `Signature` to `PaymentChannelSignature` to be more explicit.

Let's explain this in a comment.